modules/ROOT/pages/design-migrating-ms.adoc

```diff
@@ -0,0 +1,108 @@
+= Migrate to the Version 2 of the Mocking Service
+
+The mocking service lets you test endpoints that are defined in an API specification. The advantage to using the mocking service is that you do not need to implement the API before testing. To test an endpoint, the mocking service simulates calls to its methods.
+
+For example, when you are designing an API specification in API Designer, you can simulate a call to an endpoint’s GET method to see whether the response body for the 200 status code is as you want it. As a further example, if you are evaluating whether you want to implement an API specification that is in Exchange, you can simulate a call to an endpoint’s GET method to find out what the response body looks like.
+
+[IMPORTANT]
+====
+After January 31, 2020, version 1 of the mocking service will no longer send responses to requests. If you are using a third-party tool to send requests, the mocking service will return the HTTP status code 301 and provide a link to this document.
+====
+
+== How to Migrate an API-Specification Project to Version 2 of the Mocking Service
+
+If an API-specification project is older than January 10, 2019 and the mocking service was switched on before that date, follow these steps to migrate to version 2 of the mocking service:
+
+. In the notification that appears in the top-right corner after you open the project, click *Upgrade*. The text of the notification reads “A new mocking service is available. Upgrading takes just seconds.”
++
+*Result:* The mocking service is switched off and then on. The format of the base URI that the mocking service generates now differs from the format of the previous base URI.
++
+Version 1 of the mocking service used base URIs that began with `https://mocksvc.mulesoft.com`.
++
+Version 2 uses base URIs that begin with `https://anypoint.mulesoft.com/mocking/api/v1/links/`.
+
+. If any third-party tools are testing endpoints in the API specification for that project, replace the previous base URI with the new base URI in the links that those tools are using.
```

Comment:
The link id will also change after switching off/on mocking service, so the user not only has to change the base URI but need to change the link id as well

Reply:
@dfmacias So, in version 1, the link followed directly after https://mocksvc.mulesoft.com/?

Reply:
In version 1, a mock URL looks like this:
`https://mocksvc.mulesoft.com/mocks/473749b1-8b67-44bd-a572-3b362b34fb89`

after migrating this mock from APID, the URL will look like this:
`https://anypoint.mulesoft.com/mocking/api/v1/links/512f2bef-3efd-4678-bf43-9f96abab65b2/`

so, the user should replace the first URL with the second one, as the mock id changes.

What do you mean when saying `base URI`? do you mean the whole URL as shown in the examples above? If that the case, it is correct

Reply:
@dfmacias , by "base URI" I mean the URI that the mocking service appends to when generating the URIs for methods.
For example, this is the base URI for a spec:
ttps://anypoint.mulesoft.com/mocking/api/v1/links/7d19e624-a0d6-4d77-9fb8-8c78a8aeb575/api
For this particular GET method on the /customers endpoint, this is the URI:
https://anypoint.mulesoft.com/mocking/api/v1/links/7d19e624-a0d6-4d77-9fb8-8c78a8aeb575/api/customers

Customers really need to replace only the base URI, not the full URI.

Reply:
I totally agree. I just made the comment because it was not clear to me what you meant with `base URI`

it is clearer now with the example. Just one last comment, the link id will change, so if you had this link

`https://mocksvc.mulesoft.com/mocks/7d19e624-a0d6-4d77-9fb8-8c78a8aeb575/api/customers`

the id `7d19e624-a0d6-4d77-9fb8-8c78a8aeb575` will be changed with a new one, for instance

`https://anypoint.mulesoft.com/mocking/api/v1/links/512f2bef-3efd-4678-bf43-9f96abab65b2/api/customers`

Reply:
Got it. I've changed the ID from MS2 in the example that I added to the doc.